src/components/Carousel.vue

```diff
@@ -0,0 +1,276 @@
+            @focusin="!disabled && _updateSelection(entry)">
+            <slot :name="entry"></slot>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Watch, Vue } from 'vue-property-decorator';
+import SmallPage from './SmallPage.vue';
+import { Tweenable } from '@nimiq/utils';
+
+@Component({components: {SmallPage}})
+export default class Carousel extends Vue {
+    @Prop({
+        type: Array,
+        default: () => [],
+        validator: (entries: any) => Array.isArray(entries)
+            && entries.length > 0
+            && !entries.some((entry) => typeof entry !== 'string'),
+    })
+    public entries!: string[];
+
+    @Prop(String)
+    public selected?: string;
+
+    @Prop({
+        type: Number,
+        default: 16,
+    })
+    public entryMargin!: number;
+
+    @Prop({
+        type: Number,
+        default: 1000,
+    })
+    public animationDuration!: number; // in ms
+
+    @Prop({
+        type: Boolean,
+        default: false,
+    })
+    public hideBackgroundEntries!: boolean;
+
+    @Prop({
+        type: Boolean,
+        default: false,
+    })
+    public disabled!: boolean;
+
+    public $refs: { [ref: string]: HTMLElement[] }; // these are arrays because of v-for
+
+    private effectiveSelected: string = '';
+    private radius: Tweenable = new Tweenable();
+    private rotations: Map<string, Tweenable> = new Map(); // map entry -> rotation
+    private requestAnimationFrameId: number | null = null;
+
+    private get _hasDummyPosition(): boolean {
+        // add dummy to avoid that second entry is hidden exactly behind selected item on opposite side of circle.
+        return this.entries.length <= 2;
+    }
+
+    private get _totalPositionCount(): number {
+        return this.entries.length + (this._hasDummyPosition ? 1 : 0);
+    }
+
+    private async mounted() {
+        this._onKeydown = this._onKeydown.bind(this);
+        document.addEventListener('keydown', this._onKeydown);
+        // trigger these manually instead of via immediate watcher to avoid animating on first render
+        await this._updateDimensions(false);
+        this._updateSelection(this.selected);
+        this._updateRotations(false);
+    }
+
+    private destroyed() {
+        document.removeEventListener('keydown', this._onKeydown);
+        if (this.requestAnimationFrameId === null) return;
+        cancelAnimationFrame(this.requestAnimationFrameId);
+    }
+
+    @Watch('entries')
+    private async _onEntriesChange() {
+        await this._updateDimensions();
+        this._updateSelection(this.effectiveSelected); // revalidate
+        this._updateRotations();
+    }
+
+    @Watch('selected')
+    private _updateSelection(newSelection) {
+        const oldSelection = this.effectiveSelected;
+        const isNewSelectionValid = this.entries.indexOf(newSelection) !== -1;
+        const isOldSelectionValid = this.entries.indexOf(oldSelection) !== -1;
```

Comment:
I think by now it's safe to use `Array.includes()`.

Reply:
It's not supported by the tsconfig this PR is based on.
However, I upgraded this in the `Timer` PR, so I'll change it here too.